tests/Cache/CacheClientTest.php

```diff
@@ -933,4 +933,224 @@ public function testDictionaryDeleteHappyPath()
         $response = $this->client->dictionaryGet($this->TEST_CACHE_NAME, $dictionaryName, $field);
         $this->assertNotNull($response->asMiss());
     }
+
+    public function testDictionaryFetchWithNullCacheNameIsError()
+    {
+        $this->expectException(TypeError::class);
+        $cacheName = null;
+        $dictionaryName = uniqid();
+        $this->client->dictionaryFetch($cacheName, $dictionaryName);
+    }
+
+    public function testDictionaryFetchWithNullDictionaryNameIsError()
+    {
+        $this->expectException(TypeError::class);
+        $dictionaryName = null;
+        $this->client->dictionaryFetch($this->TEST_CACHE_NAME, $dictionaryName);
+    }
+
+    public function testDictionaryFetchMissingHappyPath()
+    {
+        $dictionaryName = uniqid();
+        $response = $this->client->dictionaryFetch($this->TEST_CACHE_NAME, $dictionaryName);
+        $this->assertNotNull($response->asMiss());
+    }
+
+    public function testDictionaryFetchHappyPath()
+    {
+        $dictionaryName = uniqid();
+        $field1 = uniqid();
+        $field2 = uniqid();
+        $value1 = uniqid();
+        $value2 = uniqid();
+        $contentDictionary = [$field1 => $value1, $field2 => $value2];
+        $response = $this->client->dictionarySet($this->TEST_CACHE_NAME, $dictionaryName, $field1, $value1, true, 10);
+        $this->assertNotNull($response->asSuccess());
+        $response = $this->client->dictionarySet($this->TEST_CACHE_NAME, $dictionaryName, $field2, $value2, true, 10);
+        $this->assertNotNull($response->asSuccess());
+
+        $response = $this->client->dictionaryFetch($this->TEST_CACHE_NAME, $dictionaryName);
+        $this->assertNotNull($response->asHit());
+        $this->assertEquals($response->asHit()->dictionary(), $contentDictionary);
+        $this->assertSame($response->asHit()->dictionary(), $contentDictionary);
```

Comment:
I don't think we need this `assertSame`, but let me know if you had a particular reason for adding it.

Reply:
That's true. We're not asserting to check two objects are equal and reference the same object, so I will remove it.